internal/dispatch/graph/graph.go

```diff
@@ -24,23 +24,23 @@ const errDispatch = "error dispatching request: %w"
 var tracer = otel.Tracer("spicedb/internal/dispatch/local")
 
 // NewLocalOnlyDispatcher creates a dispatcher that consults with the graph to formulate a response.
-func NewLocalOnlyDispatcher() dispatch.Dispatcher {
+func NewLocalOnlyDispatcher(concurrencyLimit uint16) dispatch.Dispatcher {
 	d := &localDispatcher{}
 
-	d.checker = graph.NewConcurrentChecker(d)
+	d.checker = graph.NewConcurrentChecker(d, concurrencyLimit)
 	d.expander = graph.NewConcurrentExpander(d)
-	d.lookupHandler = graph.NewConcurrentLookup(d, d)
+	d.lookupHandler = graph.NewConcurrentLookup(d, d, concurrencyLimit)
 	d.reachableResourcesHandler = graph.NewConcurrentReachableResources(d)
 
 	return d
 }
 
 // NewDispatcher creates a dispatcher that consults with the graph and redispatches subproblems to
 // the provided redispatcher.
-func NewDispatcher(redispatcher dispatch.Dispatcher) dispatch.Dispatcher {
-	checker := graph.NewConcurrentChecker(redispatcher)
+func NewDispatcher(redispatcher dispatch.Dispatcher, concurrencyLimit uint16) dispatch.Dispatcher {
```

Comment:
Is there a dispatch test of some kind we can add to verify the limit works as intended?

Reply:
I could definitely test the `dispatchAllAsync` with a unit test. Would that work? Hard to test it from the dispatch layer because the behavior is supposed to be identical/opaque.

Reply:
Yeah, that's good

Reply:
Added